### PR TITLE
Log Redaction: #3297 auth log redaction

### LINKED ISSUE #3297: Log redaction: 'auth' package
## 'auth' package log redaction

- Find all logging calls - use regex: `base\.(Log(f|To|Error|Panic|Fatal)?|Warn|TEMP|Error)\(`
- Identify ones that have redactable data and wrap with `base.UD()`, `base.MD()` or `base.SD`
- Change the logging calls to use the redacted variants (`base.WarnR()`, etc.) introduced in #3295 

Could also change `%v` formatting directives to the correct types (i.e. `%s`, `%b`, `%d`)

### PR DESCRIPTION
**Pull request by bbrks**
Closes #3297 - Use log redaction in auth package

### CODE REVIEW

File: auth/auth.go [hunk 1]
```diff
@@ -291,31 +291,31 @@ func (auth *Authenticator) AuthenticateUser(username string, password string) Us
 
 // Authenticates a user based on a JWT token string and a set of providers.  Attempts to match the
 // issuer in the token with a provider.
-// Used to authenticate a JWT token coming from an insecure source (e.g. client request)
+// Used to authenticate a JWT token coming from an insecure source (e.g. § request)
```
Comment (tleyden) [inline auth/auth.go]: §?

File: auth/oidc.go [hunk 2]
```diff
@@ -217,27 +217,27 @@ func (op *OIDCProvider) FetchCustomProviderConfig(discoveryURL string) (*oidc.Pr
...
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		base.LogTo("OIDC+", "Error invoking calling discovery URL %s: %v", discoveryURL, err)
+		base.LogToR("OIDC+", "Error invoking calling discovery URL %s: %v", base.UD(discoveryURL), err)
 		return nil, err
 	}
 	defer resp.Body.Close()
 	if err := json.NewDecoder(resp.Body).Decode(&customConfig); err != nil {
-		base.LogTo("OIDC+", "Error parsing body %s: %v", discoveryURL, err)
+		base.LogToR("OIDC+", "Error parsing body %s: %v", discoveryURL, err)
```
Comment (tleyden) [inline auth/oidc.go]: Looks like the other cases of `discoveryURL` were wrapped in `base.UD()`

**Review by tleyden — CHANGES_REQUESTED** [review]
See inline

File: auth/oidc.go (hunk 2 above)
Comment (bbrks) [reply auth/oidc.go]: Thanks, good spot!

File: auth/auth.go (hunk 1 above)
Comment (bbrks) [reply auth/auth.go]: No idea how that made it in there! Fixed

**Review by tleyden — APPROVED** [review]

**Pull request merged**